server/src/controllers/TagsController.ts

```diff
@@ -0,0 +1,132 @@
+import { Router, Request, Response } from "express";
+import Tag from "../models/Tag";
+import Controller from "../interfaces/ControllerInterface";
+
+/**
+ * The tags controller is responsible for handling the HTTP requests.
+ * Examples would be GET, POST, PUT, DELETE.
+ */
+class TagsController implements Controller {
+  // Path that is required in order to access the api http://localhost:8080/routes/api/tags
+  public path = "/api/tags";
+  public router = Router();
+
+  constructor() {
+    this.initializeRoutes();
+  }
+
+  /**
+   * Creates the routes for the Tag Controller
+   * Ex. GET, PUT, POST, UPDATE, etc
+   */
+  public initializeRoutes(): void {
+    this.router.route(this.path).get(this.getAllTags).post(this.createTag);
+    this.router
+      .route(this.path + "/:id")
+      .get(this.getTag)
+      .put(this.updateTag)
+      .delete(this.deleteTag);
+  }
+
+  // Goes to route /api/tags
+
+  /**
+   * Grabs all tags in the database and sends them as a response in json
+   * @param request HTTP browser request
+   * @param response HTTP browser response
+   */
+  getAllTags = async (request: Request, response: Response): Promise<void> => {
```

Comment:
We most likely won't need to return all of the Tags from the database at once, this is the only function that may not be used. Everything else looks good so long as it was tested and verified working.